app/src/main/java/com/rkm/tasky/database/dao/EventDao.kt

```diff
@@ -0,0 +1,24 @@
+package com.rkm.tasky.database.dao
+
+import androidx.room.Dao
+import androidx.room.Delete
+import androidx.room.Query
+import androidx.room.Upsert
+import com.rkm.tasky.database.model.EventEntity
+import com.rkm.tasky.database.model.EventWithDetails
+
+@Dao
+interface EventDao {
+
+    @Query("SELECT * FROM events, attendees, photos where id = :id")
+    suspend fun getAllEventDetails(id: String): EventWithDetails
+
+    @Query("SELECT * FROM events where id = :id")
+    suspend fun getEventById(id: String)
+
+    @Upsert
+    suspend fun upsertEvent(event: EventEntity)
+
+    @Delete
+    suspend fun deleteEvent(event: EventEntity)
```

Comment:
For simplicity in your code, I would suggest you pass in the ID only and not the entire model. Room will, under the hood, also use the field marked with `@PrimaryKey` to find the event to delete 😂 So needing to pass the entire model to delete it is just an extra responsibility with not value and you’ll always have the ID of an event, but not always the entire model.